File: Binary_Search/069.Sqrtx/069.Sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if (x <= 1) {
            return x;
        }

        int low = 0;
        int high = x;
        int ret = 0;
        while (low <= high) {
            int middle = low + (high - low) / 2;
            if (middle <= x / middle) {
                ret = middle;
                low = middle + 1;
            } else {
                high = middle - 1;
            }
        }
        return ret;
    }
}

File: Binary_Search/069.Sqrtx/069.Sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        low = 0
        high = x

        while low < high:
            middle = math.ceil((low + high) / 2) #high - (high - low) // 2
            if middle * middle <= x :
                low = middle
            else:
                high = middle - 1
        return low
